cmake/OpenCVFindOpenEXR.cmake

```diff
@@ -144,6 +144,11 @@ IF(OPENEXR_FOUND)
   endif()
   if(NOT OPENEXR_VERSION)
     SET(OPENEXR_VERSION "Unknown")
+  else()
+    if(HAVE_CXX17 AND OPENEXR_VERSION STRLESS "2_3")
+      message(STATUS "  OpenEXR(ver ${OPENEXR_VERSION}) doesn't support C++17 and higher. Updating OpenEXR 2.3.0+ is required.")
+      SET(OPENEXR_FOUND FALSE)
```

Comment:
This may be a fix for 4.x but definitely not for https://github.com/opencv/opencv/issues/26673 or 5.x. Compile standard of 5.x has been switched to C++17.

Reply:
Yes, and this problem is existed not only 5.x branch but also 4.x branch.

I think after accepting pull request for 4.x branch, it will be merged into 5.x branch.

https://github.com/opencv/opencv/wiki/How_to_contribute#before-you-start-contributing-you-should

> If you are going to fix a bug, check that it still exists. This can be done by building the latest [4.x](https://github.com/opencv/opencv/tree/4.x)/[5.x](https://github.com/opencv/opencv/tree/5.x) branch or the [latest master branch](https://github.com/opencv/opencv), and make sure that the error is still reproducible there.

Reply:
> I think after accepting pull request for 4.x branch, it will be merged into 5.x branch.

Yes, I understand that. But my point is 5.x is now based on C++17 and integrated OpenEXR is 2.3.0 which supports C++17, so solution for 5.x should be:

1. Detect system OpenEXR version.
2. If the system one is below the integrated one, which is 2.3.0, the integrated one should be enabled; that is to build with flag `-DBUILD_OPENEXR=ON`.
3. If the system one is equal to or higher than the integrated one, build should go with the system one.

Now you are basically turning off building with OpenEXR, right?

Reply:
I'm sorry but it is diffuclt to agree about this change request for me.

Even If system OpenEXR is older than v2.3.0 it doesn't means OpenCV user requests to use bundled OpenEXR(it is slightly old too). If needed, we can set enable with `-DBUILD_OPENEXR=ON` manually.

Forcing ‘BUILD_OPENEXR`=`ON` losts user-selectability and it looks like bug.

Related: https://github.com/opencv/opencv/issues/21326

Reply:
> Forcing ‘BUILD_OPENEXR=ON` losts user-selectability and it looks like bug.

I am quite open to this. But for your solution, you should add a message saying bundled OpenEXR is available for building 5.x.